### ISSUE TITLE
BETA! support for NULL string & binary properties

### ISSUE THREAD

**Issue by heardrwt** [realm/realm-cocoa#628]
This would allow for the storage of prop = [NSData data] and prop = nil;

**Comment by ishaq** [comment]
hope it gets integrated soon

**Comment by alazier** [comment]
We have someone actively working on this, but we need to add support for each type individually in core, so this work will take quite a bit of time to complete.

**Comment by yasirmturk** [comment]
sanitizing json is a load on performance and memory

**Comment by timanglade** [comment]
I know @rrrlasse is working on this in core at the moment, but it would be good to add a subsection about this + workaround to the [Current Limitations](realm.io/docs/cocoa#current-limitations) section of our docs right away. /cc @jpsim @segiddins

**Comment by checkbutton** [comment]
@timanglade @jpsim Are there any updates on this issue? Missing support for null values broke our app so badly, that we had to switch back to Core Data. :-(

**Comment by jpsim** [comment]
> @timanglade @jpsim Are there any updates on this issue? Missing support for null values broke our app so badly, that we had to switch back to Core Data. :-(

We hear you, null support is a top priority for us. We're adding in null support type by type in Realm's core database engine. Strings are done, and progress on other types is advancing steadily. We'll expose that in the higher level language bindings as soon as it's ready.

In the meantime, it really shouldn't be necessary to completely move away from Realm for this limitation. Our current suggested workaround should be flexible enough to support any scenario: wrap properties that should be nullable in an RLMObject subclass:

```objc
@interface NullableInt : RLMObject
@property NSInteger intValue;
@end
```

**Comment by timanglade** [comment]
Let me just pile up here on what @jpsim is saying and add that yes, we’ve
had someone working on this full-time for weeks now, but not allowing any
nullable fields was such a core design decision, that it affected
everything from how we store data on disk, to how queries are handled, and
so this hasn’t been an easy fix at all. It is close to being done though,
and we’ll try to release support for the most obvious types (string and
data) asap. Our apologies again for it taking so long.

**Comment by yasirmturk** [comment]
+1 @checkbutton same story

**Comment by mootoh** [comment]
@jpsim, That workaround looks good but that does not allow properties to be populated properly from JSON object. Suppose we have the following JSON received from API server:

```
{
  "name": "Me",
  "age": 30
}
```

and hope to populate it in Realm:
```
@interface NullableInt : RLMObject
@property NSInteger intValue;
@end

@interface Client : RLMObject
@property NullableInt age;
@end
```

Calling `[[Client alloc] initWithObject:...]` raises an exception...

**Comment by Mazyod** [comment]
Just wanted to share my experience regarding this situation:

Especially since I am using Swift, I need to handle `nil` cases explicitly when accessing the variables, anyways, meaning property being set to `nil` won't reduce the code written. Sure it will be brilliant to have that, but since we don't, might as well utilize the [Null Object Pattern](http://sourcemaking.com/design_patterns/null_object).

It takes a bit more extra work to decide what value would represent nil, but in my case, empty strings and `-1`s did the trick. Granted this is not ideal nor universal, but it has the advantage of easily upgradable when nil support is rolled in, since the property type remains intact.

Now, be creative how you want to check for that Null Object. Perhaps an extension on the data type itself (remember, it works for `Int`, `Double`, and all Swift types, basically), or an operator, or whatever.

I guess this is more of the elephant in the room, but [I learned to introduce him](https://www.goodreads.com/quotes/76530-when-there-s-an-elephant-in-the-room-introduce-him) if he was ever around.

**Comment by mikecole20** [comment]
@mootoh, I can't get that workaround to work either. The problem is that Realm then expects a dictionary for ```NullableInt``` that looks like ```{"intValue" = 5}```, but we really just have a ```5``` there because it is an Int, not another object.

@jpsim, do you have a solution that still allows for parsing the JSON correctly without having to manually alter the JSON first? If we are going to that trouble, we may as well just create the objects manually.

**Comment by jpsim** [comment]
Those are your options, @mikecole20. You could also use something like [Realm-JSON](https://github.com/matthewcheok/Realm-JSON) that would do it for you.

**Comment by fnxpt** [comment]
Do you have an ETA when the core will support NULL values.

**Comment by timanglade** [comment]
We will start beta-testing null support for strings in a matter of days
(will circulate details about that here, on Twitter and on the Google Group
for people who are interested). General availability and more types will
follow.

**Comment by reejosamuel** [comment]
@timanglade using realm-json still gives the same error. Is there is some settings we need to configure for it to work?

**Comment by timanglade** [comment]
@reejosamuel You should ask for help as a new issue on https://github.com/matthewcheok/Realm-JSON We try to keep this thread 100% about the null issue since many people are following this issue to know when we will release it.

**Comment by 3lvis** [comment]
Hey guys, we've faced this problem too, at the moment we handled this issue by preprocessing (mincing) our JSON so it fits better with Realm.

https://github.com/hyperoslo/Minced

[In case you're interested we added an example showing how this works.]
(https://github.com/hyperoslo/Minced#example)

**Comment by cfeckardt** [comment]
@timanglade Any news on beta-testing?

**Comment by timanglade** [comment]
Yes, just finished the package :) Here we go! We’re starting slow with a release that hopefully will _not_ do anything at all. We’ve tested it internally as much as we can, your help in trying it out is much appreciated so we can start releasing betas that actually add functionality asap.

/cc @tgoyne @rrrlasse @simonask @segiddins @bmunkholm

# Realm Core Null v1 BETA

THIS IS A BETA RELEASE — USE IT AT YOUR OWN RISK. It might delete your data or cause other catastrophic events. Please report any bugs or issues on Github ([Cocoa](https://github.com/realm/realm-cocoa/issues), [Java](https://github.com/realm/realm-javaa/issues))  and clearly mention you are using the “null beta”.



## Release Notes

**Realm Core Null v1 does not expose new features to you**. What it does is silently upgrade our file format to enable null support. This is a critical step for us to validate. Without confirmation from a number of beta-testers, we will not be able to release Realm Core Null v2, which will let you have null string properties (more types will follow).

You can find the core release that includes null support at https://static.realm.io/downloads/core/realm-core-null-v1.zip



## Instructions


### Realm Cocoa

#### If you are building Realm Cocoa from source

1. Build https://github.com/realm/realm-cocoa from master
2. Replace the library files under core-0.89.0 with those contained in this zip
3. Rebuild and use the resulting framework in your file

#### If you are using Realm through CocoaPods

1. Replace the files under Pods/Realm/core with the contents of the core folder of this zip

#### If you are using the zip releases from our website

You will not be able to test null support at this time. We may release beta binaries with null support at a later date. In the meantime you can build an equivalent .framework from source using the instructions above.



## Help!

Please report any bugs or issues on Github ([Cocoa](https://github.com/realm/realm-cocoa/issues), [Java](https://github.com/realm/realm-javaa/issues))  and clearly mention you are using the “null beta”.

**Comment by matti777** [comment]
Cool, anxiously waiting for this to be part of the distribution. Luckily the deadline for our project (for which I decided to adopt Realm.io last week..) is still a few months away. :)

**Comment by bmunkholm** [comment]
PLEASE also report if you are using this without problems! The fater we get it tested, the faster we can release the real thing :-)

**Comment by jpsim** [comment]
> Cool, anxiously waiting for this to be part of the distribution.

You can try it right away, instructions are in https://github.com/realm/realm-cocoa/issues/628#issuecomment-94050752.

**Comment by shmidt** [comment]
@jpsim I tried to build static framework with null support, but the problem is that after adding new core ```build.sh``` script downloads old core.
Can you please put here compiled frameworks with null support?

**Comment by jpsim** [comment]
@shmidt the zip included `release_notes.txt`, which `build.sh` uses to check if it has the appropriate version of core. So I just renamed it to `releasenotes.txt` and re-uploaded the zip, so copying those files should now work.

Please remember that this release has no feature changes, so you won't be able to save null strings. It's rather a beta that tests out the automatic file format upgrade required for null functionality.

**Comment by anlaital** [comment]
Doesn't look like the file format upgrade is working as expected.

I have a scenario where I have defined a primary key for my objects and populated the objects using an earlier Realm version (0.91.3). When I upgrade to the new format, it seems that the objects in question are being duplicated and primary key constraints are not being respected. This results in the database containing two identical objects that also have the same primary key.

**Comment by segiddins** [comment]
@anlaital I was able to reproduce the issue you're seeing, and I'm going to dive in and investigate -- thanks so much for trying out the upgrade and reporting this!

**Comment by enricenrich** [comment]
One question @timanglade, which values now support null? All the values?

**Comment by timanglade** [comment]
None at the moment, per the release notes. This first beta just tests the
file format upgrade. The next beta will introduce support for null strings.

**Comment by kitasuke** [comment]
This beta is only available with 0.91.2 or later, right? Because the libtightdb-ios.a file name had changed to librealm-ios since 0.91.2. I think you should mention that on the release note above. Someone might want to use it with older version!

**Comment by timanglade** [comment]
@kitasuke Thanks! I updated the note.

**Comment by martinprot** [comment]
I'm using Cocoapods. I just downloaded the .zip and replaced the Pods/Realm/core files, as you described. Should I include them in the Xcode bundle?
When I compile, i've got some undefined symbols. Did I miss something?

**Comment by jpsim** [comment]
Hi @martinprot, sorry for the trouble, the instructions in the README and the GitHub comment above are a bit over-simplified for CocoaPods. The best way to fully explain how to do this is to show you, so I recorded a quick screencast demonstrating how to use the null beta of core in Realm installed with CocoaPods: http://youtu.be/imfgNqIuT14.

Please let us know if you have any further questions.

**Comment by martinprot** [comment]
Thanks a lot for your reactivity @jpsim ! Now it's clear <3

**Comment by ArthurBelmonte** [comment]
Realm Core Null v1 BETA upgrade worked fine with me.

Now we wait for the null string support.....

**Comment by timanglade** [comment]
We heard some positive reports so we will proceed with the next phase shortly; we’re a bit caught up in the release of Realm Swift now but will make progress on this front soon…

**Comment by komocode** [comment]
Cool, can't wait.

**Comment by segiddins** [comment]
# Realm Objective-C & Realm Swift Null Beta v2

THIS IS A BETA RELEASE — USE IT AT YOUR OWN RISK. It might delete your data or cause other catastrophic events. Please report any bugs or issues on [GitHub](https://github.com/realm/realm-cocoa/issues) and clearly mention you are using `null-string-beta-2`.

## Release Notes

Realm Objective-C & Realm Swift Null Beta v2 allows you to make string and binary properties nullable.

In Realm Objective-C, `NSString` and `NSData` properties are optional by default, but you can override `+[RLMObject requiredProperties:]` to make a string or binary property required.

In Realm Swift, we use the native Swift `Optional<T>` type to determine whether a property should be made optional or not. One caveat to this is that optional string properties (`String?`) must actually be declared as `NSString?` due to [a Swift runtime bug](http://openradar.appspot.com/21143551).

## Instructions

### If you are building Realm from source

1. Build https://github.com/realm/realm-cocoa from the `null-string-beta-2` tag
2. Use the resulting framework(s) in your project (the instructions should be the same as https://realm.io/docs/swift/latest/ or https://realm.io/docs/objc/latest/)

### If you are using Realm through CocoaPods

```ruby
# Podfile
pod 'Realm', :git => 'https://github.com/realm/realm-cocoa', :tag => 'null-string-beta-2'

# Uncomment the following line if you're using RealmSwift
# pod 'RealmSwift', :git => 'https://github.com/realm/realm-cocoa', :tag => 'null-string-beta-2'
```

### If you are using the zip releases from our website

We are not offering prebuilt binaries for this beta. We may release beta binaries with null support at a later date. In the meantime you can build an equivalent `.framework` from source using the instructions above.

## Help!

Please report any bugs or issues on [Github](https://github.com/realm/realm-cocoa/issues) and clearly mention you are using the “null beta”.

**Comment by tariq235** [comment]
Will this solve NSDate nil issue too ? I am unable to store nil in NSDate.

**Comment by segiddins** [comment]
The current beta does not support nil dates, but it's definitely on the roadmap!

-Samuel E. Giddins

On Jun 3, 2015, at 10:29 PM, Tariq Mohammad <notifications@github.com> wrote:

Will this solve NSDate nil issue too ? I am unable to store nil in NSDate.

**Comment by bmunkholm** [comment]
@tariq235 @komocode @martinprot @ArthurBelmonte @kitasuke @enricenrich @anlaital @shmidt @matti777 @mikecole20 @yasirmturk @fnxpt @reejosamuel @3lvis 
Any chance you guys have had the chance to try out the beta? We would be grateful for some feedback!

Thanks!

**Comment by komocode** [comment]
I'm currently re-doing models for a random app to use Realm, I'll give the beta a shot. I'll report back soon!

**Comment by shmidt** [comment]
@bmunkholm After installation of ```pod 'RealmSwift', :git => 'https://github.com/realm/realm-cocoa', :tag => 'null-string-beta-2'```

I got compiler error: ```Pods/RealmSwift/RealmSwift/Property.swift:46:40: 'RLMProperty' does not have a member named 'optional'```

**Comment by tgoyne** [comment]
You need to add `pod 'Realm', :git => 'https://github.com/realm/realm-cocoa', :tag => 'null-string-beta-2'` to your podspec in addition to the RealmSwift one.

**Comment by shmidt** [comment]
@tgoyne Oh, I see. Thanks, will test it again.

**Comment by shmidt** [comment]
@bmunkholm @tgoyne Worked great for me, no data was lost, no bugs were noticed.

**Comment by anlaital** [comment]
@bmunkholm I will roll out an internal test release with this today as well. I will make sure to report any and all findings to you guys.

**Comment by scottfister** [comment]
I am attempting to do this with a float or double and I get the error `'NSUnknownKeyException', reason: '[<__NSCFNumber 0x7fe509f350e0> valueForUndefinedKey:]: this class is not key value coding-compliant for the key float.'`.

My code:
```
class NullableFloat: Object {
    dynamic var float: Float = 0.0
}
class Foo: Object {
  dynamic var bar : NullableFloat?
}
```

What am I doing wrong?

**Comment by segiddins** [comment]
Are you trying to set the nullable float to an NSNumber instead of a NullableFloat object?

-Samuel E. Giddins

On Jun 11, 2015, at 6:11 PM, Scotty <notifications@github.com> wrote:

I am attempting to do this with a float or double and I get the error 'NSUnknownKeyException', reason: '[<__NSCFNumber 0x7fe509f350e0> valueForUndefinedKey:]: this class is not key value coding-compliant for the key float.'.

My code:

class NullableFloat: Object {
    dynamic var float: Float = 0.0
}
class Foo: Object {
  dynamic var bar : NullableFloat?
}
What am I doing wrong?

**Comment by tariq235** [comment]
Realm Objective-C Null Beta v2 upgrade worked fine for me.
Added a workaround for NSDate with [NSDate dateWithTimeIntervalSince1970:1]

**Comment by bmunkholm** [comment]
Just to be clear: The only NULL supported properties are strings and data. We are working on adding  support for integer, date and float/double types. Those still have to be wrapped in a class as workaround.

**Comment by scottfister** [comment]
@segiddins I'm using the Realm REST API method to automatically create the records: `realm.create(Foo.self, value: foo, update: true)`. Is there a way to continue using this method and have the nullable property work too? The convenience method saves a *lot* of manual mapping, as I have 3 levels of nested relationships it's automatically creating.

**Comment by fnxpt** [comment]
Crashing with the following exception
Terminating app due to uncaught exception 'RLMException', reason: 'No value or default value specified for property 'tokenForFacebook' in 'NBUser''

**Comment by jpsim** [comment]
Could you send us a project that exhibits this behavior, @fnxpt so we can investigate further? Thanks!

**Comment by fnxpt** [comment]
@jpsim I will try to get a sample project with it... meanwhile can you tell me how can I see if the migration was successful, it looks like the fields are not optional.

**Comment by jpsim** [comment]
If a migration block completes, it means it was run successfully.

**Comment by jessyconroy** [comment]
Crashing with the following exception
Terminating app due to uncaught exception 'RLMException', reason: 'No value or default value specified for property 'userDescription' in 'DBUserProfile'' where userDescription is an optional String?

**Comment by segiddins** [comment]
@jessyconroy only the null beta supports nulls, are you using the beta?

**Comment by jessyconroy** [comment]
@segiddins Nope. I was not...I thought it was updated with the newest release. Will give the beta a whirl now. Thanks

**Comment by jessyconroy** [comment]
I'm now getting an error "Terminating app due to uncaught exception 'RLMException', reason: 'Invalid property'"

**Comment by segiddins** [comment]
Is it a `String?` property? If so, it will need to be declared as `NSString?`. If that's not the issue, could you please share a reproduction example, either here or privately at help@realm.io so I can investigate? Thanks!

**Comment by jessyconroy** [comment]
Nope...its not a String?. I'll email you the classes. Thanks

**Comment by jessyconroy** [comment]
@segiddins Email sent! Thanks for your help :+1:

**Comment by jessyconroy** [comment]
@segiddins Thanks a million for helping me resolve the issue. It was an error on my part in the end. Cheers again! :+1:

**Comment by fnxpt** [comment]
@jpsim , sorry for the delay, I just sent you an email with a sample project that shows the crash.

**Comment by fnxpt** [comment]
@jpsim any news about the crash?

**Comment by timanglade** [comment]
JP’s out for a bit. Did you send the project to help@realm.io? If so someone else can help you while he is away. Sorry about that!

**Comment by fnxpt** [comment]
@timanglade just forward the email to help@realm.io

**Comment by segiddins** [comment]
@fnxpt thanks for the email! I took a look at the project you sent, and saw that we had a branch with the same name as the tag, which interfered with CocoaPods' downloader (since tags and branches are both named refs under the hood in git).
You should be able to fix the problem by running:

```
rm -rf ~/Library/Caches/CocoaPods/Pods
rm -rf Pods
pod install
```

**Comment by fnxpt** [comment]
@segiddins Thanks, after removing the Pod files I was able to updated and run it without the crash that I was reporting, but I got another one.
Terminating app due to uncaught exception 'RLMException', reason: 'No value or default value specified for property 'currency' in 'NBFavorite''

I think the problem is related with Arrays, because if I add a single object it works

**Comment by fnxpt** [comment]
Actually if I update an object I don't get the initial crash, but if I use addOrUpdate it crashes on NSStrings and NSData

**Comment by fnxpt** [comment]
Forget my last comments, it looks like I was having an issue with cocoa pods cache.
The only crash that Im having now is this:
Terminating app due to uncaught exception 'RLMException', reason: 'Unable to automatically copy a property of type 'data' to type 'data''

**Comment by fnxpt** [comment]
Well, after creating a new database it works, migration did not work, it fails when migrating NSData objects and if i put them as requiredProprieties the app hangs in the migration process.
With a new database its working until now

**Comment by segiddins** [comment]
@fnxpt I think I have since fixed the last issue you described, but if you could send us another version of the project that demonstrates it, I'd love to verify it has indeed been fixed. Thanks!

**Comment by triage** [comment]
hey there.
yes, I'm on the Beta.
I've got an interesting issue I'm having a hard time sorting through... here's what my model looks like
```swift
class MyExample: Object {
    dynamic var typeString: NSString?
```
... leads to an RLM exception: `Terminating app due to uncaught exception 'RLMException', reason: 'Only 'object' properties can be made optional, and property 'typeString' is of type 'string'.'`

**Comment by segiddins** [comment]
@triage that would imply you're running Realm from master, rather than the beta tag?

**Comment by lokielse** [comment]
How About add `<Optional>, <Ignore> ` protocols to property? just like https://github.com/icanzilb/JSONModel
```
@interface ProductModel : JSONModel
@property (assign, nonatomic) int id;
@property (strong, nonatomic) NSString<Optional>* name;
@property (assign, nonatomic) float price;
@property (strong, nonatomic) NSNumber<Optional>* uuid;
@end
```

**Comment by bcapps** [comment]
:+1: Excited for this.

**Comment by bcapps** [comment]
@segiddins I'm curious, does the segfault still happen in Swift 2?

**Comment by jpsim** [comment]
> I'm curious, does the segfault still happen in Swift 2?

Was there ever a segfault? A few people commented on this thread that their apps were aborting due to uncaught exceptions when assigning null values prior to the beta (which is by design). Realm won't throw an exception when assigning null to an optional string or data property if using the null beta, if that's what you're asking.

**Comment by bcapps** [comment]
Ah, should have clarified, I meant the `String` segfault that was happening in the runtime and required the use of `NSString`.

**Comment by segiddins** [comment]
I think it might've been fixed in 2.0, but I'm not sure

**Comment by jpsim** [comment]
Ah, I see! In Swift 2.0 (and even in Swift 1.2 IIRC), setting `String?` (with realm's null beta release) should work, as well as `NSString?`.

**Comment by bcapps** [comment]
Interesting. Well, if there's a tag or branch available somewhere of this feature beta with RealmSwift and Swift 2.0, I'd be happy to test `String?` support.

**Comment by jpsim** [comment]
We're tracking adding that functionality in #2308. @bcapps is there a reason you have to use `String?` rather than `NSString?` given that Swift will bridge one to the other?

**Comment by bcapps** [comment]
@jpsim No functional reason, however I can't test either currently because my project uses Swift 2.0 and this beta tag was created a couple months ago, before the RealmSwift 2.0 changes. I was offering to test if you had such a build, but I see that you've updated your comment to reflect that it has not been released publicly.

**Comment by remotevision** [comment]
:+1: glad to see this one is being test! I'd be happy to test as well.

**Comment by shawnwall** [comment]
What additional testing needs to be completed for the obj-c version to get this merged in? How can I help?

**Comment by timanglade** [comment]
We are fixing a few file format bugs we found through testing and have started adding support for the remaining types (int, float and all Cocoa types that derive from that). We’re anxious to deliver this too, and it looks like you won’t have to wait much longer. We’re not fully set yet on whether we’ll a) ship just string & byte support first, or b) all types at once. The former might be confusing and require you upgrade your app’s code in multiple passes. If we pick the former, this should be released very soon as we’re pretty much just waiting on bugfixes; if the latter it will take a bit longer as we’ll need to finalize support for the new types.

**Comment by i-schuetz** [comment]
+1 for optional NSDate...

**Comment by komocode** [comment]
I'm starting on a new project. Do you imagine support for all types will make it within the next month?

**Comment by bmunkholm** [comment]
We always try to stay away of promising any dates of feature releases. We want to have the flexibility to prioritise bugs and support above new features, which just makes it hard to predict feature releases. That said though, the core functionality for all types is now done (in the C++ Core), so we only got left to fully integrate this into the bindings. So yeah, if support doesn't get too crazy, I expect us to release support for nullability in all types within the next month - no promises though! :-)

**Comment by komocode** [comment]
Great! No worries if it's delayed, I just needed a ballpark number.

**Comment by AndrewBarba** [comment]
Hey if anyone is interested, I ported AFNetworking's `JSONObjectByRemovingKeysWithNullValues` over to Swift 2.0 and it's been working really well for us:
```
public static func JSONObjectByRemovingKeysWithNullValues(JSONObject: AnyObject) -> AnyObject {
        switch JSONObject {
        case let JSONObject as [AnyObject]:
            return JSONObject.map { JSONObjectByRemovingKeysWithNullValues($0) }
        case let JSONObject as [String: AnyObject]:
            var mutableDictionary: [String: AnyObject] = [:]
            for (key, value) in JSONObject {
                switch value {
                case _ as NSNull:
                    mutableDictionary.removeValueForKey(key)
                case _ as [AnyObject], _ as [String: AnyObject]:
                    mutableDictionary[key] = JSONObjectByRemovingKeysWithNullValues(value)
                default:
                    mutableDictionary[key] = value
                }
            }
            return mutableDictionary
        default:
            return JSONObject
        }
    }
```

**Comment by segiddins** [comment]
Support for nullable dates, ints, floats, doubles, and bools is being actively worked upon as I type this -- there's a pull request up at https://github.com/realm/realm-cocoa/pull/2458 that adds this functionality for both Swift & Objective-C.

**Comment by AndrewBarba** [comment]
Awesome thanks for the update

**Comment by siuying** [comment]
@segiddins Any way I can NULL String + Swift 2? Or I should use #2458 ?

**Comment by segiddins** [comment]
@siuying not yet, unfortunately. You won't be able to use #2458 yet because it requires a new release of our C++ core.

**Comment by siuying** [comment]
@segiddins thanks, I'd wait :)

**Comment by robertjpayne** [comment]
:+1: this is a huge enhancement and looking forward to it!

**Comment by pimnijman** [comment]
While we're waiting... here's my temporarily solution (in Swift). I'm hoping to easily migrate this to optionals when Realm is ready for it. 

Instead of

    dynamic public var myProperty: String?

I'll be using

    dynamic public var myPropertyIsNil = true
    dynamic public var myPropertyValue = ""
    public var myProperty: String? {
        get {
            return (self.myPropertyIsNil) ? nil : self.myPropertyValue
        }
        set {
            self.myPropertyIsNil = (newValue == nil)
            self.myPropertyValue = newValue ?? ""
        }
    }

Be sure to add "myProperty" to your ignoredProperties array.

**Comment by jessyconroy** [comment]
Hi,

Any update on a NULL values Realm release for Swift 2.0? Xcode 7 went live the other day and I migrated my code from Swift 1.2 to 2.0. I'd say I'm not the only person who updated :)

**Comment by pravintate** [comment]
realm swift 0.95.2 version 
how to set Int double variable as optional. I don't want to set zero at declaration time as follow
dynamic var count : Int = 0

**Comment by kishikawakatsumi** [comment]
@pravintate 0.95.2 and 0.95.3 does not support any null values yet. So you should use the flag that indicate a value is null or wrap the value with `Object`. e.g. https://github.com/realm/realm-cocoa/issues/1614#issuecomment-78567104

Please wait releasing future version that support null values.

**Comment by pravintate** [comment]
@kishikawakatsumi: If I set flag as per your suggestion then I will increase unnecessary coding for me and in few days Realm will support null that time again I need to update my code as per that. so using flag instead of null is not good option.

**Comment by timothyrobb** [comment]
What they are saying, is that if you need support now, use a flag, like many others have until now. Or just wait. They were responding to your question.

**Comment by pravintate** [comment]
Ok thanks for replying

**Comment by jpsim** [comment]
We've officially released Realm Objective-C & Swift 0.96 with support for null values, keypath collection queries, RealmCollectionType & more! Read about it here: https://realm.io/news/realm-objc-swift-0.96.0/

Thank you all for your patience!

**Comment by yasirmturk** [comment]
awesome changes the way we live :100:
